front-end/src/app/layout/committee-banner/committee-banner.component.ts

```diff
@@ -6,24 +6,99 @@ import { Subject } from 'rxjs';
 @Component({
   selector: 'app-committee-banner',
   templateUrl: './committee-banner.component.html',
-  styleUrls: ['./committee-banner.component.scss']
+  styleUrls: ['./committee-banner.component.scss'],
 })
 export class CommitteeBannerComponent implements OnInit, OnDestroy {
-  committeeName: string | undefined;
-  subBannerItems: (string | undefined)[] = [];
+  committeeName?: string;
+  committeeStatus?: string;
+  committeeFrequency?: string;
+  committeeType?: string;
+  committeeID?: string;
   private destroy$ = new Subject<boolean>();
 
-  constructor(private store: Store) { }
+  constructor(private store: Store) {}
+
+  getFilingFrequency(frequency_code: string): string {
+    switch (frequency_code) {
+      case 'T':
+        return 'Terminated';
+      case 'A':
+        return 'Administratively terminated';
+      case 'D':
+        return 'Active - Debt';
+      case 'W':
+        return 'Active - Waived';
+      case 'M':
+        return 'Active - Monthly filer';
+      case 'Q':
+        return 'Active - Quarterly filer';
+    }
+    return '';
+  }
+
+  public getCommitteeActive(frequency_code: string): boolean {
+    switch (frequency_code) {
+      case 'M':
+      case 'Q':
+      case 'W':
+      case 'D':
+        return true;
+    }
+    return false;
+  }
+
+  public getCommitteeType(committee_type_code: string): string {
+    switch (committee_type_code) {
```

Comment:
This could very concisely be represented as a map

```
const committeeCodeLabels = {
    C: 'communication cost',
    D: 'delegate',
    E: 'electioneering communication'
    ...
```

and accessed like: 

```
return this.committeeCodeLabels[code] || '';
```

Reply:
That's a much, much better idea, lol